docs/source/howto/codes.rst

```diff
@@ -3,173 +3,200 @@
+   That is why we used the key-notation in the ``define`` method.
+
+Finally, we pass the |CodeInfo| to a |CalcInfo| object (one CalcJob can involve more than one executable, so ``codes_info`` is a list).
+We define the ``retrieve_list`` of filenames that the engine should retrieve from the directory where the job ran after it has finished.
 The engine will store these files in a :py:class:`~aiida.orm.nodes.data.folder.FolderData` node that will be attached as an output node to the calculation with the label ``retrieved``.
 There are :ref:`other file lists available<topics:calculations:usage:calcjobs:file_lists>` that allow you to easily customize how to move files to and from the remote working directory in order to prevent the creation of unnecessary copies.
 
-This was a minimal example of how to implement the |CalcJob| class to interface AiiDA with an external code.
-For more detailed information and advanced functionality on the |CalcJob| class, refer to the Topics section on :ref:`defining calculations <topics:calculations:usage>`.
+This was an example of how to implement the |CalcJob| class to interface AiiDA with an external code.
+For more details on the |CalcJob| class, refer to the Topics section on :ref:`defining calculations <topics:calculations:usage>`.
 
 .. _how-to:codes:parsing:
 
 Parsing the outputs
 ===================
 
-The parsing of the output files generated by a |CalcJob| is optional and can be used to store (part of) their information as AiiDA nodes, which makes the data queryable and therefore easier to access and analyze.
-To enable |CalcJob| output file parsing, one should subclass the |Parser| class and implement the :py:meth:`~aiida.parsers.parser.Parser.parse` method.
-The following is an example implementation, as shipped with ``aiida-core``, to parse the outputs of the :py:class:`~aiida.calculations.arithmetic.add.ArithmeticAddCalculation` discussed in the previous section:
+Parsing the output files produced by your code into AiiDA nodes is optional, but it can make your data queryable and therefore easier to access and analyze.
 
-.. literalinclude:: ../../../aiida/parsers/plugins/arithmetic/add.py
-    :language: python
-    :pyobject: ArithmeticAddParser
+We start our ``parser.py`` plugin by subclassing the |Parser| class and implementing its :py:meth:`~aiida.parsers.parser.Parser.parse` method.
+
+.. code-block:: python
+
+    class ArithmeticAddParser(Parser):
+        """Parser for an `ArithmeticAddCalculation` job."""
+
+        def parse(self, **kwargs):
+            """Parse the contents of the output files stored in the `retrieved` output node."""
+            output_folder = self.retrieved
+
+            with output_folder.open(self.node.get_option('output_filename'), 'r') as handle:
+                result = int(handle.read())
+
+            self.out('sum', Int(result))
```

Comment:
I see the point that you want to first show the minimal version without exit codes, but the downside of this method is that now the code is not tested. If we change anything to the code, this might easily get broken without us noticing. We should consider if this is worth the risk and if it is really that bad to not have the full version of the code. I think people are capable of overlooking the exit codes until we get to them a bit further on

Reply:
The alternative is that I also add this simple parser to the codebase and test it.

Reply:
I'm adding the code into the codebase + adding a test